File: sums.rb
require 'json'

package = JSON.parse(File.read("package.json"))
version = package["version"]

files = [
  "Standard Notes-#{version}-mac.zip",

  "Standard Notes-#{version}.dmg",
  "Standard Notes-#{version}.dmg.blockmap",

  "Standard Notes-#{version}-i386.AppImage",
  "Standard Notes-#{version}.AppImage",

  "Standard Notes Setup #{version}.exe",
  "Standard Notes Setup #{version}.exe.blockmap",

  "standard-notes_#{version}_amd64.snap",

  "latest-linux-ia32.yml",
  "latest-linux.yml",
  "latest-mac.yml",
  "latest.yml"
]

output = ""

files.each do |file|
  sum = `cd dist && sha256sum '#{file}'`
  output += "#{sum}"
end

File.write('dist/SHA256SUMS', output)

puts "Successfully wrote SHA256SUMS:\n#{output}"

File: scripts/sums.js
const crypto = require('crypto');
const fs = require('fs');

function sha256(filePath) {
  return new Promise((resolve, reject) => {
    fs.createReadStream(filePath)
      .pipe(crypto.createHash('sha256').setEncoding('hex'))
      .on('finish', function () {
        resolve(this.read());
      })
      .on('error', reject);
  });
}

(async () => {
  console.log('Writing SHA256 sums to dist/SHA256SUMS');

  try {
    const version = JSON.parse(fs.readFileSync('./package.json')).version;
    const files = [
      `standard-notes-${version}-mac.zip`,
      `standard-notes-${version}-mac.dmg`,
      `standard-notes-${version}-mac.dmg.blockmap`,

      `standard-notes-${version}-linux-i386.AppImage`,
      `standard-notes-${version}-linux-x86_64.AppImage`,
      `standard-notes-${version}-linux-amd64.snap`,

      `standard-notes-${version}-win.exe`,
      `standard-notes-${version}-win.exe.blockmap`,

      'latest-linux-ia32.yml',
      'latest-linux.yml',
      'latest-mac.yml',
      'latest.yml',
    ];

    process.chdir('dist');

    let hashes = await Promise.all(
      files.map(async (fileName) => {
        const hash = await sha256(fileName);
        return `${hash}  ${fileName}`;
      })
    );
    hashes = hashes.join('\n');
    await fs.promises.writeFile('SHA256SUMS', hashes);
    console.log(`Successfully wrote SHA256SUMS:\n${hashes}`);
  } catch (err) {
    console.error(err);
    process.exitCode = 1;
  }
})();
